src/acceptance-test/java/org/zalando/nakadi/webservice/timelines/TimelineSyncAT.java

```diff
@@ -0,0 +1,201 @@
+        } finally {
+            publishing.close();
+        }
+        int attemptsLeft = 5;
+        while (attemptsLeft-- > 0) {
+            // Wait a little bit for lock release
+            Thread.sleep(300);
+            if (updated.get()) {
+                break;
+            }
+        }
+        Assert.assertEquals(true, updated.get());
+    }
+
+    @Test
+    public void testPublishPauseOnTimelineUpdate() throws InterruptedException, IOException {
+        final TimelineSync t1 = createTimeline();
+        final TimelineSync t2 = createTimeline();
+        final String eventType = UUID.randomUUID().toString();
+        // Lock publishing
+        t1.startTimelineUpdate(eventType, TimeUnit.SECONDS.toMillis(30));
+        final AtomicBoolean lockTaken = new AtomicBoolean(false);
+        new Thread(() -> {
+            try (Closeable cl = t2.workWithEventType(eventType, TimeUnit.SECONDS.toMillis(5))) {
+                lockTaken.set(true);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }, "publisher").start();
+
+        Thread.sleep(TimeUnit.SECONDS.toMillis(2)); // Wait for thread to start.
+        Assert.assertEquals(false, lockTaken.get());
+
+        // Now release event type
+        t1.finishTimelineUpdate(eventType);
+        Assert.assertEquals(true, lockTaken.get());
+    }
+
+    private static class RefreshListener {
+        private final Map<String, Integer> data;
+
+        private RefreshListener() {
+            this.data = new HashMap<>();
+        }
+
+        public Map<String, Integer> getData() {
+            return data;
+        }
+
+        public Consumer<String> createConsumer() {
+            return this::accept;
+        }
+
+        public void accept(final String s) {
+            if (!data.containsKey(s)) {
+                data.put(s, 1);
+            } else {
+                data.put(s, data.get(s) + 1);
+            }
+        }
+    }
+
+    @Test
+    public void testNotificationsCalledWhenTimelineUpdated() throws InterruptedException {
+        final String eventType1 = UUID.randomUUID().toString();
+        final String eventType2 = UUID.randomUUID().toString();
+
+        final RefreshListener l1 = new RefreshListener();
+        final TimelineSync t1 = createTimeline();
+        t1.registerTimelineChangeListener(eventType1, l1.createConsumer());
+
+        final RefreshListener l2 = new RefreshListener();
+        final TimelineSync t2 = createTimeline();
+        t2.registerTimelineChangeListener(eventType2, l2.createConsumer());
+
+        t1.startTimelineUpdate(eventType1, TimeUnit.SECONDS.toMillis(3));
+        t1.startTimelineUpdate(eventType2, TimeUnit.SECONDS.toMillis(3));
+
+        Assert.assertEquals(0, l1.getData().size());
+        Assert.assertEquals(0, l2.getData().size());
+
+        t1.finishTimelineUpdate(eventType1);
+        // Wait for listeners
+        Thread.sleep(TimeUnit.SECONDS.toMillis(1));
```

Comment:
I would suggest replacing `sleep` call by retries whenever possible. From our experience it leads to a more stable test suite.